File: 56-MergeIntervals/56-MergeIntervals.py
# Last updated: 8/19/2026, 10:41:23 AM
class Solution:
    def merge(self, intervals):
        """
        Merges overlapping intervals in a list of intervals.
        Args:
            intervals: A list of intervals, where each interval is a list of two integers [start, end].
        Returns:
            A list of non-overlapping intervals.
        """
        intervals.sort(key=lambda x: x[0])  
        merged = []
        start, end = intervals[0]

        for i in range(1, len(intervals)):
            interval_start, interval_end = intervals[i]
            if interval_start <= end:  
                end = max(end, interval_end)
            else:  
                merged.append([start, end])
                start, end = interval_start, interval_end
        merged.append([start, end])

        return merged

File: 56-MergeIntervals/56-MergeIntervals.java
// Last updated: 8/19/2026, 10:41:17 AM
class Solution {
    public int[][] merge(int[][] arr) {
        Arrays.sort((arr),(a,b)->a[0]-b[0]);
        List<int[]> res = new ArrayList<>();
        int start = arr[0][0];
        int end = arr[0][1];
        for(int i=0;i<arr.length;i++)
        {
            if(end>=arr[i][0])
            end = Math.max(end,arr[i][1]);
            else
            {
                res.add(new int[]{start,end});
                start=arr[i][0];
                end = arr[i][1];
            }
        }
        res.add(new int[]{start,end});
        return res.toArray(new int[res.size()][]);
    }
}